gradle-sls-packaging/src/test/groovy/com/palantir/gradle/dist/ProductDependencyLockFileTest.groovy

```diff
@@ -25,13 +25,13 @@ class ProductDependencyLockFileTest extends Specification {
         when:
         List<ProductDependency> sample = [
                 new ProductDependency("com.palantir.product", "foo", "1.20.0", "1.x.x", null),
-                new ProductDependency("com.palantir.other", "bar", "0.2.0", "0.x.x", null)
+                new ProductDependency("com.palantir.other", "bar", "0.2.0", "0.x.x", null, true)
         ]
 
         then:
         ProductDependencyLockFile.asString(sample, [] as Set<ProductId>, '0.0.0') == """\
         # Run ./gradlew --write-locks to regenerate this file
-        com.palantir.other:bar (0.2.0, 0.x.x)
+        com.palantir.other:bar (0.2.0, 0.x.x) optional
```

Comment:
@ferozco Made a good point to me that this will break our pdeps-bot, which needs to be updated to handle optional deps. The file should be the same until someone adds an optional dep, then pdeps-bot will error out for that PR, so I don't think that should block this PR, but we need to fix that too.

Reply:
I am not familiar with them, but I can look into fixing them if you can point me to them after this PR